# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/fs/api/FileSystemSyncAccessHandle.h
# Commit: b68e19021833
# Full Hash: b68e1902183357e129a1a53490447fc3cc219f4f
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2023-02-07 16:42:31
# Regressor Bug: 1809064
# File Overlap Count: 1
# Description:
#   Bug 1809064 - Add PFileSystemAccessHandleControl top level protocol; r=dom-storage-reviewers,jari
#   
#   Sync access handle methods need to run sync loops to block the thread until a
#   result is available. The main protocol PFileSystemManager and its subprotocol
#   PFileSystemAccessHandle is bound to the main worker event queue, so incoming
# ==============================================================================

diff -r fc3ab5ee8ef3 -r b68e19021833 dom/fs/api/FileSystemSyncAccessHandle.h
--- a/dom/fs/api/FileSystemSyncAccessHandle.h	Tue Feb 07 08:21:15 2023 +0000
+++ b/dom/fs/api/FileSystemSyncAccessHandle.h	Tue Feb 07 08:27:06 2023 +0000
@@ -23,6 +23,7 @@
 namespace dom {
 
 class FileSystemAccessHandleChild;
+class FileSystemAccessHandleControlChild;
 struct FileSystemReadWriteOptions;
 class FileSystemManager;
 class MaybeSharedArrayBufferViewOrMaybeSharedArrayBuffer;
@@ -39,6 +40,8 @@
       mozilla::ipc::RandomAccessStreamParams&& aStreamParams,
       mozilla::ipc::ManagedEndpoint<PFileSystemAccessHandleChild>&&
           aAccessHandleChildEndpoint,
+      mozilla::ipc::Endpoint<PFileSystemAccessHandleControlChild>&&
+          aAccessHandleControlChildEndpoint,
       const fs::FileSystemEntryMetadata& aMetadata);
 
   NS_DECL_CYCLE_COLLECTING_ISUPPORTS
@@ -48,6 +51,8 @@
 
   void ClearActor();
 
+  void ClearControlActor();
+
   bool IsOpen() const;
 
   bool IsClosing() const;
@@ -86,6 +91,7 @@
       nsIGlobalObject* aGlobal, RefPtr<FileSystemManager>& aManager,
       mozilla::ipc::RandomAccessStreamParams&& aStreamParams,
       RefPtr<FileSystemAccessHandleChild> aActor,
+      RefPtr<FileSystemAccessHandleControlChild> aControlActor,
       RefPtr<TaskQueue> aIOTaskQueue,
       const fs::FileSystemEntryMetadata& aMetadata);
 
@@ -104,6 +110,8 @@
 
   RefPtr<FileSystemAccessHandleChild> mActor;
 
+  RefPtr<FileSystemAccessHandleControlChild> mControlActor;
+
   RefPtr<TaskQueue> mIOTaskQueue;
 
   nsCOMPtr<nsIRandomAccessStream> mStream;